Misc/NEWS.d/next/Core_and_Builtins/2024-12-15-16-56-26.gh-issue-126085.BvyEXk.rst

```diff
@@ -0,0 +1,2 @@
+Add the ``tp_iter`` slot to :class:`typing.TypeAliasType` to allow star
```

Comment:
I'm with Bénédikt, I think we shouldn't explicitly mention the C-API-level type slot here. If we really want to mention the magic method, let's use the Python variation of it:
```suggestion
Add the :meth:`~object.__iter__` slot to :class:`typing.TypeAliasType` to allow star
```

Reply:
Ah, I misunderstood what Bénédikt meant :laughing: I'll leave out the method reference here as well.